core/import_data/import_admb.py

```diff
@@ -0,0 +1,486 @@
+                "string": string_data["StringText"],
+                "using_cfc": "using cfc" in string_data["LangDefault"].lower(),
+            }
+
+    return strings
+
+
+def get_parent_index(index):
+    """
+    Get parent index from index string
+        e.g. "1.2.1" -> "1.2"
+             "1.1" -> "1"
+             "1" -> ""
+    @param index = str (index)
+
+    @return parent_index = str
+    """
+    parent_index = index.split(".")
+    parent_index.pop()
+    parent_index = ".".join(parent_index)
+
+    return parent_index
+
+
+def set_type(articles_dict):
+    """
+    Set type for each article
+    ! we need type to create adm rows correctly
+    @param articles_dict = dict (articles dictionary)
+    """
+    for index in articles_dict:
+        parent_index = articles_dict[index]["parent_index"]
+        # set type
+        if parent_index and parent_index in articles_dict:
+            # if the type is not set for the article, set it to question
+            articles_dict[index]["type"] = articles_dict[index].get(
+                "type", AdmRow.AdmRowType.QUESTION
+            )
+            # parent type should be subtitle or title
+            if articles_dict[parent_index].get("type") == AdmRow.AdmRowType.QUESTION:
+                articles_dict[parent_index]["type"] = AdmRow.AdmRowType.SUBTITLE
+        else:
+            # for top level articles set type to title
+            articles_dict[index]["type"] = AdmRow.AdmRowType.TITLE
+
+
+def parse_admb_file(file_name, strings_dict):
+    """
+    Parse admb json file and create a dictionary
+      - based on strings dictionary we set the text for each article
+      - if the article is not in strings dictionary, we add it to articles_without_text
+    ! we need this dictionaries to create adm rows
+
+    @param file_name = str (file path for import file)
+    @param strings_dict = dict (strings dictionary)
+
+    @return articles_dict, articles_without_text = tuple(dict, dict)
+        - articles_dict struct: {
+            index: {
+                "text": admb_text,
+                "sort_order": admb_sort_order,
+                "cp_id": admb_cp_id,
+                "parent_index": admb_parent_index
+                "index": admb_index
+                "using_cfc": admb_using_cfc (bool)
+            }
+        }
+        - articles_without_text struct: {
+            ArticleId: {
+                "type": AdmRow.AdmRowType.USER_TEXT,
+                "sort_order": admb_sort_order,
+                "cp_id": admb_cp_id,
+                "parent_index": admb_parent_index
+                "index": admb_index
+            }
+    """
+
+    with open(file_name, "r", encoding="utf8") as f:
+        articles_json = json.load(f)
+
+    articles = {}
+    articles_without_text = {}
+    for article in articles_json:
+        index = article["IndexId"].strip(".")
+        article_data = {
+            "sort_order": article["SortOrder"],
+            "cp_id": article["Id"],
```

Comment:
Pentru ce avem nevoie de asta?

Reply:
nu stiu ma gandeam ca vrem sa avem log-uri sa stim exact ce si de unde s-a importat. Dar daca nu trebuie pot sterge partea asta

Reply:
aici am nevoie de el ulterior ca sa gasesc itemurile in dictionar, dar in model nu e nevoie de el asa ca l-am sters in pr-ul cu admc